pilot/pkg/bootstrap/sidecarinjector.go

```diff
@@ -57,3 +92,93 @@ func (s *Server) initSidecarInjector(args *PilotArgs) error {
 	log.Infof("Skipping sidecar injector, template not found")
 	return nil
 }
+
+const delayedRetryTime = time.Second
+
+// Moved out of injector main. Changes:
```

Comment:
See https://github.com/istio/istio/pull/19008, I did something similar but didn't duplicate the code which may be a bit more maintainable

Reply:
Or I guess it may make sense to if we are making these modifications

Reply:
I don't think it'll be more maintainable - same with starting up grpc listener. 
A bit of duplication with clear methods is better a complex method with hard to understand behavior. Besides - the old one will go away.